File: https:/github.com/Amrit-raj50/neetcode-submissions/7-ReverseInteger/7-ReverseInteger.py
# Last updated: 16/06/2026, 14:13:48
class Solution:
    def reverse(self, x: int) -> int:
        dup=abs(x)
        rev_x=0
        while dup>0:
            dig=dup%10
            rev_x=rev_x*10+dig
            dup=dup//10
        
        if x<0:
            rev_x=rev_x* -1
            
        if rev_x < -2**31 or rev_x > 2**31 - 1 :
            return 0

        return rev_x

File: https:/github.com/Amrit-raj50/neetcode-submissions/7-ReverseInteger/7-ReverseInteger.cpp
// Last updated: 04/08/2026, 23:17:06
class Solution {
public:
    int reverse(int x) {
        if(x < INT_MIN || x > INT_MAX){
            return 0;
        }
        bool flag = false;
        if(x < 0){
            flag = true;
            if(x == INT_MIN){
                return 0;
            }
            x = x * -1;
        }

        int temp = x;
        int sum = 0;
        int dig;
        bool trace = false;

        while(temp > 0){
            dig = temp % 10;
            temp = temp / 10;
            if(sum  > (INT_MAX - dig) / 10){
                sum = 0;
                break;
            }
            sum = sum * 10 + dig;
        }
        if(flag){
            sum = sum * -1;
        }

        return sum;
    }
};